immunopepper/preprocess.py

```diff
@@ -362,189 +362,198 @@ def parse_gene_metadata_info(h5fname, sample_list, cross_graph_expr):
-        logging.error("Sample {} not found in the h5 variant file".format(mutation_sample))
+    if mutation_sample and mutation_sample not in file_sample_list:
+        logging.error(f'Sample {mutation_sample} not found in the h5 variant file')
         sys.exit(1)
-    col_id = [i for (i, item) in enumerate(file_sample_list) if item == mutation_sample][0]
-    row_id = np.where(np.logical_or(a['gt'][:,col_id] == heter_code,a['gt'][:,col_id] == 1))[0]
+    col_id = file_sample_list.index(mutation_sample)  # TODO: what if the index is not found?
+    # the 'gt' column stores the most likely genotype for the sample
+    row_id = np.where(np.logical_or(a['gt'][:, col_id] == heter_code, a['gt'][:, col_id] == 1))[0]
     for irow in row_id:
-        chromo = encode_chromosome(a['pos'][irow,0])
-        pos = a['pos'][irow,1]-1
+        chromosome = encode_chromosome(a['pos'][irow, 0])
+        pos = a['pos'][irow, 1] - 1
         mut_base = decode_utf8(a['allele_alt'][irow])
         ref_base = decode_utf8(a['allele_ref'][irow])
-        var_dict = {"mut_base":mut_base,"ref_base":ref_base}
-        if (mutation_sample,chromo) in mut_dict:
-            mut_dict[(mutation_sample,chromo)][pos] = var_dict
+        var_dict = {'mut_base': mut_base, 'ref_base': ref_base}
+        if (mutation_sample, chromosome) in mut_dict:
+            mut_dict[(mutation_sample, chromosome)][pos] = var_dict
         else:
-            mut_dict[(mutation_sample,chromo)] = {}
-            mut_dict[(mutation_sample,chromo)][pos] = var_dict
-    logging.info("TEST {}".format(len(mut_dict)))
+            mut_dict[(mutation_sample, chromosome)] = {pos: var_dict}
     return mut_dict
 
 
-def parse_mutation_from_maf(mutation_tag, target_sample_list, mutation_sample, maf_path, output_dir='', mut_pickle=False, sample_eq_dict={}):
-    """
-    Extract somatic mutation information from given maf file.
+def _check_mutation_samples(mutation_sample: str, target_samples: list[str], sample_eq_dict: dict[str, str],
+                            loaded_samples: set[str]):
+    """ Asserts that the mutation samples loaded from a VCF or MAF file contain mutation_sample. """
+    for target_sample in target_samples:
+        if mutation_sample == target_sample and sample_eq_dict[target_sample] not in loaded_samples:
+            logging.error(f'Samples in mutation file are {loaded_samples}')
+            logging.error(f'No mutations for sample "{sample_eq_dict[target_sample]}" found, '
+                          f'please check samples above or consider using --sample-name-map')
+            sys.exit(1)
 
-    Parameters
-    ----------
-    maf_path: str, maf file path
-    output_dir: str, save a pickle for maf_dict to save preprocess time
-    mut_pickle: bool, flag indicating whether to pickle mutation info to disk
 
-    Returns
-    -------
-    mut_dict: with key (sample, chromo) and values (var_dict)
+def _check_mutation_samples_new(mutation_sample: str, loaded_samples: set[str]):
```

Comment:
Is this a suggestion to replace `_check_mutation_samples`? Currently, it is not referenced from anywhere. In any case, the translation via `sample_eq_dict` is only there for historic reasons, where the sample names in SplAdder runs and mutation files were inconsistent. Moving forward, we can just introduce the requirement that they are the same and let the user fix the inputs.

Reply:
Thinking about this, it might not be so straightforward, as mutations are usually called on different samples (e.g., whole exome sequencing) than the ones the splice events were called from (RNA-Seq). But one can still require the user to fix this and use something like the patient or sample donor ID per sample.